spring-cloud-dataflow-composed-task-runner/src/main/java/org/springframework/cloud/dataflow/composedtaskrunner/support/UnexpectedTaskExecutionException.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2017-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.composedtaskrunner.support;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.springframework.batch.core.UnexpectedJobExecutionException;
+import org.springframework.boot.ExitCodeGenerator;
+import org.springframework.cloud.task.repository.TaskExecution;
+
+/**
+ * Creates a {@link UnexpectedTaskExecutionException} which extends {@link UnsupportedOperationException}, but
+ * also contains the exitCode as information.
+ *
+ * @author Tobias Soloschenko
+ */
+public class UnexpectedTaskExecutionException extends UnexpectedJobExecutionException implements ExitCodeGenerator {
+
+	private static final long serialVersionUID = 3494615323781500165L;
+
+	/**
+	 * The unique id associated with the task execution.
+	 */
+	private long executionId;
```

Comment:
Any reason not to just hold a reference to the TaskExecution since you are capturing all of its fields?

Reply:
I did so because of two reasons:
1. The TaskExecution itself is not serializable and I didn't wanted to change this
2. This way you can decide which information are available in the exception

Reply:
> This way you can decide which information are available in the UnexpectedTaskExecutionException

Right, but currently we are adding all fields. How about we have a compromise and start out w/ a handful of essential fields (not including the parameters :) ) maybe these:

These for identifying the task execution:
```
executionId = taskExecution.getExecutionId();
externalExecutionId = taskExecution.getExternalExecutionId();
taskName = taskExecution.getTaskName();
```

These for identifying the problem:
```
errorMessage = taskExecution.getErrorMessage();
exitCode = taskExecution.getExitCode();
exitMessage = taskExecution.getExitMessage();
```

Wdyt?

Reply:
For parameter I think this is to much you are right but I would include start and end time as well. 👍

Reply:
If you can update the commit w/ this change I can merge. Otherwise I can make the change and merge in the next 24hrs. Either way works for me. Thanks @klopfdreh

Reply:
Hey @onobc - arguments are removed - the parentExecutionId I also kept in the class as this might be also relevant to get a hint from which parent the actual task has been started.